docs/lang/articles/contribution/contributor_guide.md

```diff
@@ -4,296 +4,324 @@ sidebar_position: 1
-  triggering CI. For example, a commit with the message
-  `[skip ci] This commit will not trigger CI` will not trigger CI.
-- A tick on the left-hand side of a commit hash means CI passed, while
-  a cross means CI failed.
-
-## Enforcing code style
-
-- Locally, you can run `python misc/code_format.py` in the command line to re-format
-  code style. Note that you have to install `clang-format-10` and
-  `yapf v0.31.0` locally.
-
-- If you don't have these formatting tools locally, feel free to
-  leverage GitHub Actions: simply comment `/format` in a PR
-  (e.g., [#2481](https://github.com/taichi-dev/taichi/pull/2481#issuecomment-872226701))
-  and then [Taichi Gardener](https://github.com/taichi-gardener)
-  will automatically push a commit to your branch that formats the code.
-  Note if you want to make more changes afterwards, you'll need to
-  `git pull` first.
-
-- For your C++ code, please also follow [C++ style](./cpp_style).
-
-## PR title format and tags
-
-PR titles will be part of the commit history reflected in the `master`
-branch, therefore it is important to keep PR titles readable.
-
-- Please always prepend **at least one tag** such as `[Lang]` to PR
-  titles:
-  - When using multiple tags, make sure there is exactly one
-    space between tags;
-  - For example, `[Lang][refactor]` (no space) should be replaced
-    by `[Lang] [refactor]`.
-- The first letter of the PR title body should be capitalized:
-  - For example, `[Doc] improve documentation` should be replaced by
-    `[Doc] Improve documentation`;
-  - `[Lang] "ti.sqr(x)" is now deprecated` is fine because `"`
-    is a symbol.
-- Please do not include back quotes ("`") in PR titles.
-- Good examples include `[Metal] Support bitmasked SNode`, `[Vulkan]
-  ti.atomic_min/max support`, or `[Opt] [ir] Enhanced intra-function optimizations`.
+- Prepend at least one tag, such as `[Lang]`, to your PR title:
+  - When it comes to multiple tags, separate adjacent tags with exactly one space;
+- `[Lang][refactor]` (no space) should be replaced by `[Lang] [refactor]`.
+- Capitalize the initial letter of the word immediately after your tag(s):
+  -	`[Doc] improve documentation` should be replaced by `[Doc] Improve documentation`;
+  -	`[Lang] "ti.sqr(x)" is now deprecated` is fine because `"` is a symbol.
+
+- Do not include back quotes ("\`") in the title of a PR.
+- **Good examples** 
+  - `[Metal] Support bitmasked SNode`, 
+  - `[Vulkan]ti.atomic_min/max support`, or 
+  - `[Opt] [ir] Enhanced intra-function optimizations`.
 
 Frequently used tags:
 
-- `[CPU]`, `[CUDA]`, `[Metal]`, `[Vulkan]`, `[OpenGL]`: backends;
-- `[LLVM]`: the LLVM backend shared by CPUs and CUDA;
-- `[Lang]`: frontend language features, including syntax sugars;
-- `[Std]`: standard library, e.g., `ti.Matrix` and `ti.Vector`;
-- `[Sparse]`: sparse computation;
-- `[IR]`: intermediate representation;
+- `[CPU]`, `[CUDA]`, `[Metal]`, `[Vulkan]`, `[OpenGL]`: Backends;
```

Comment:
Let's just keep "CI", "CUDA", "IR", "Lang", "Refactor".

Reply:
@ailzhang Please confirm if we need CPU, LLVM, Metal, Vulkan, OpenGL, Std, Opt, and GUI, **and** if we need "misc" if anything does not fit into the existing categories.

Reply:
Yea let's just keep the most common ones here :D